R/read_BINXLOG2R.R

```diff
@@ -0,0 +1,304 @@
+#'@title Import Risø BINX-data from the BINX Log File
+#'
+#'@description
+#' The function attempts to extract meaningful measurement data from the
+#' log file (`.txt`) stored along with a BINX file if this option was chosen.
+#' This is particularly helpful if the BINX file is broken and the data would
+#' be lost otherwise.
+#'
+#'@details
+#' This function is basically a hack trying to make the most out of a broken file in case
+#' at least the log file is salvageable. Please do not expect black magic; the extract is
+#' very basic, and the function was written to work for TL, OSL, IRSL curves only.
+#' If it breaks, it breaks.
+#'
+#' **TL curve extraction**
+#' Since the BIN/BINX do not contain x-values, they also cannot be created magically
+#' here. To avoid problems with the conversion, we assume always BIN-file version `3` for
+#' the conversion. This should be okay in most cases, however, it may become problematic
+#' if you have data for dedicated curve analysis.
+#'
+#'@param file [character] (**required**): ASCII log file to read. There is no check
+#'the file extension; just see whether it works or not.
+#'
+#'@param verbose [logical] (*with default*): enable/disable output to the terminal.
+#'
+#'@param ... further arguments that will be passed to the function (currently not used)
+#'
+#'@seealso [Luminescence::read_BIN2R], [Luminescence::write_R2BIN], [Luminescence::Risoe.BINfileData-class],
+#' [base::readLines], [Luminescence::RLum.Analysis-class], [list.files]
+#'
+#'@return
+#' Returns an S4 [Luminescence::RLum.Analysis-class] object or a [list] of it.
+#'
+#'@note
+#' This function tries to extract data from a log file produced for debugging
+#' purposes. There is no guarantee regarding the format stability, and the
+#' function is meant as a last resort if your BIN/BINX files are unusable.
+#'
+#'@section Function version: 0.1.0
+#'
+#'@author
+#' Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
+#'
+#'@examples
+#'## read log file (RLum.Analysis)
+#'file <-  system.file("extdata/BINX_IRSL_LOG.TXT", package = "Luminescence")
+#'object <-  read_BINXLOG2R(file = file)
+#'
+#'## convert to RisoeBINFile-class object
+#'object <- convert_RLum2Risoe.BINfileData(object)
+#'
+#'## export as BINX
+#'\dontrun{
+#'  write_R2BIN(t, file = tempfile(), version = "04")
+#'}
+#'
+#'
+#'@keywords IO
+#'@md
+#'@export
+read_BINXLOG2R <- function(
+  file,
+  verbose = TRUE,
+  ...
+  ){
+  .set_function_name("read_BINXLOG2R")
+  on.exit(.unset_function_name(), add = TRUE)
+
+# Incoming integrity ------------------------------------------------------
+  .validate_class(file, c("character", "list"))
+  .validate_logical_scalar(verbose)
+
+# Self-call ---------------------------------------------------------------
+  if(inherits(file, "list")) {
+    out <- lapply(file, read_BINXLOG2R, verbose = verbose[1])
```

Comment:
```suggestion
    out <- lapply(file, read_BINXLOG2R, verbose = verbose)
```

We have already validated that there's only one element.